### ISSUE TITLE
Date filter converts timezone wrong

### ISSUE THREAD

**Issue by hidasandris** [metabase/metabase#11293]
Date filter at the custom question editor uses wrong timezone. I have records with date type column and setting the filter to this date returns nothing. However, setting the filter to the next day returns the records with previous day.
I have read the timezone troubleshooting guide and everything is set up correctly. The records, the database and the OS are in Europe/Budapest timezone. Custom SQL query works correctly. 
The SQL view of the question shows: 

`WHERE ("public"."sample"."date" >= '2019-09-27T00:00:00.000Z'::timestamp
AND "public"."sample"."date" < '2019-09-28T00:00:00.000Z'::timestamp)
`

**Metabase Diagnostic Info**

```{
  "browser-info": {
    "language": "hu-HU",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.93 Safari/537.36 Vivaldi/2.8.1664.40",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_181-b13",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_181",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "25.181-b13",
    "os.name": "Linux",
    "os.version": "3.16.0-4-amd64",
    "user.language": "hu",
    "user.timezone": "Europe/Budapest"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "tag": "v0.33.4",
      "date": "2019-10-07",
      "branch": "release-0.33.x",
      "hash": "9559406"
    },
    "settings": {
      "report-timezone": "Europe/Belgrade"
    }
  }
}
```

**Comment by flamber** [comment]
Hi @hidasandris 
There's a lot of work being done currently to fix those pesky timezone issues.
https://github.com/metabase/metabase/projects/9
Some of it landed in 0.33.5, so could you try that version?
Do this MySQL-specific issue #7968 look related? (I know it's the old QB design, but ignore that)
Or am I misunderstanding how you are doing filtering?

**Comment by hidasandris** [comment]
Hi @flamber 
Thank you, the 0.33.5.1 update solved the issue.
